contracts/MarginCalculator.sol

```diff
@@ -143,26 +178,39 @@ contract MarginCalculator {
      * @param _vault theoretical vault that needs to be checked
      * @return marginRequired the minimal amount of collateral needed in a vault, denominated in collateral
      */
-    function _getMarginRequired(
-        MarginVault.Vault memory _vault,
-        bool _hasShort,
-        bool _hasLong
-    ) internal view returns (FPI.FixedPointInt memory) {
-        FPI.FixedPointInt memory shortAmount = _hasShort ? FPI.fromScaledUint(_vault.shortAmounts[0], BASE) : ZERO;
-        FPI.FixedPointInt memory longAmount = _hasLong ? FPI.fromScaledUint(_vault.longAmounts[0], BASE) : ZERO;
-
-        OtokenInterface otoken = _hasShort
-            ? OtokenInterface(_vault.shortOtokens[0])
-            : OtokenInterface(_vault.longOtokens[0]);
-        bool expired = now > otoken.expiryTimestamp();
-        bool isPut = otoken.isPut();
+    function _getMarginRequired(MarginVault.Vault memory _vault, VaultDetails memory _vaultDetails)
+        internal
+        view
+        returns (FPI.FixedPointInt memory)
+    {
+        FPI.FixedPointInt memory shortAmount = _vaultDetails.hasShort
+            ? FPI.fromScaledUint(_vault.shortAmounts[0], BASE)
+            : ZERO;
+        FPI.FixedPointInt memory longAmount = _vaultDetails.hasLong
+            ? FPI.fromScaledUint(_vault.longAmounts[0], BASE)
+            : ZERO;
+
+        address otokenUnderlyingAsset = _vaultDetails.hasShort
+            ? _vaultDetails.shortUnderlyingAsset
+            : _vaultDetails.longUnderlyingAsset;
+        address otokenCollateralAsset = _vaultDetails.hasShort
+            ? _vaultDetails.shortCollateralAsset
+            : _vaultDetails.longCollateralAsset;
+        address otokenStrikeAsset = _vaultDetails.hasShort
+            ? _vaultDetails.shortStrikeAsset
+            : _vaultDetails.longStrikeAsset;
+        uint256 otokenExpiry = _vaultDetails.hasShort
+            ? _vaultDetails.shortExpiryTimestamp
+            : _vaultDetails.longExpiryTimestamp;
+        bool expired = now > otokenExpiry;
+        bool isPut = _vaultDetails.hasShort ? _vaultDetails.isShortPut : _vaultDetails.isLongPut;
```

Comment:
can you check if vaultDetails.hasShort {
// assign shortDetails
} else { 
// assign long details
} 
Does that save more gas than the check every time?

Reply:
Oh let me try that in a different branch